pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -189,6 +189,12 @@ func (r *ReconcileSyncSetInstance) Reconcile(request reconcile.Request) (reconci
 		return r.syncDeletedSyncSetInstance(ssi, ssiLog)
 	}
 
+	// Assume applied is false until we successfully apply everything to cluster.
+	ssi.Status.Applied = false
+	original := ssi.DeepCopy()
+	// Always update status if anything has changed
+	defer r.updateSyncSetInstanceStatus(ssi, original, ssiLog)
```

Comment:
This may have been lost in the rest of the discussion from the comments from @twiest, so I will reiterate that the code needs to handle properly an error returned from `updateSyncSetInstanceStatus`.

Reply:
I actually can't see where this was mentioned in the PR, was it slack perhaps? 

Can we still defer and just log the error? Or do we need to be able to requeue in the event of an error updating status, in which case this needs a rethink.

Reply:
Mentioned here.
| Also, what happens if updateSyndSetInstanceStatus errors out?

We need to re-queue in the event of an error updating the status.